aicsimageio/readers/reader.py

```diff
@@ -358,6 +388,24 @@ def size(self, dims: str = Dimensions.DefaultOrder) -> Tuple[int]:
         # Return the shape of the data for the dimensions requested
         return tuple([self.dask_data.shape[self.dims.index(dim)] for dim in dims])
 
+    def get_size_of_dimension(self, dim: str) -> int:
+        """
+        Parameters
+        ----------
+        dim: str
+            A single dimension to get the size of.
+
+        Returns
+        -------
+        size: int
+            The size of the requested dimension or 1 for empty dimension.
+        """
+        dim = dim.upper()
+        if dim in self.dims:
+            return self.dask_data.shape[self.dims.index(dim)]
```

Comment:
Are we always using dask to get the shape?

Reply:
I believe in most cases we are. Certain readers (`TiffReader` and `CziReader`) are able to get shape without loading the image data / without creating the dask array but as a default implementation this works fine imo.

Reply:
I think I should change this to be `self.shape` that way however the Reader implements `shape` it will be routed through there.

Reply:
I think I agree with that.  if ometiff reader and czi reader are getting shape without dask_data, then I feel much better.   Maybe there shouldn't even be a default implementation for this?

Reply:
For `shape` you mean?

Reply:
Resolved for default implementations of shape in various readers + improved `TiffReader` to override default implementation and read properties from metadata